python_modules/dagster/dagster/core/types.py

```diff
@@ -269,6 +281,23 @@ def __init__(
         self.is_optional = check.bool_param(is_optional, 'is_optional')
         self._default_value = default_value
 
+    @staticmethod
+    def composite_field(dagster_type, description=None):
```

Comment:
If we already want this, it should probably be the default behavior in `Field.__init__`. We could set `is_optional=sentinel` by default and make users explicitly set `is_optional=False` in order to get the strict behavior.

Reply:
This turned out to be more complicated than I thought. Will address separately. Tracking @ https://github.com/dagster-io/dagster/issues/332

Reply:
I took a crack at this just to cut my teeth on something other than the demo :)